Sources/PubNub/PubNub.swift

```diff
@@ -2034,41 +2030,17 @@ extension PubNub {
 // MARK: - PAM
 
 extension PubNub {
-  /// Perfoms a lookup and returns a token for the specified resource type and ID
-  /// - Parameters:
-  ///   - for: The resource ID for which the token is to be retrieved.
-  ///   - with: The resource type
-  /// - Returns: The assigned PAMToken if one exists; otherwise `nil`
-  ///
-  /// If no token is found for the supplied resource type and ID,
-  /// the TMS checks the resource stores in the following order: `User`, `Space`
-  public func getToken(for identifier: String, with type: PAMResourceType? = nil) -> PAMToken? {
-    return tokenStore.getToken(for: identifier, with: type)
-  }
-
-  /// Returns the token(s) for the specified resource type.
-  /// - Returns: A dictionary of resource identifiers mapped to their PAM token
-  public func getTokens(by type: PAMResourceType) -> PAMTokenStore {
-    return tokenStore.getTokens(by: type)
+  /// Extract permissions from provided token,
+  /// - Parameter token: The token from which permissions should be extracted.
+  /// - Returns: PAMToken with permissions information.
+  public func parse(token: String) -> PAMToken? {
+    return PAMToken.token(from: token)
   }
 
-  /// Returns a map of all tokens stored by the token management system
-  /// - Returns: A dictionary of resource types mapped to resource identifier/token pairs
-  public func getAllTokens() -> [PAMResourceType: PAMTokenStore] {
-    return tokenStore.getAllTokens()
-  }
-
-  /// Stores a single token in the Token Management System for use in API calls.
+  /// Stores token for use in API calls.
   /// - Parameter token: The token to add to the Token Management System.
   public mutating func set(token: String) {
-    tokenStore.set(token: token)
-  }
-
-  /// Stores multiple tokens in the Token Management System for use in API calls.
-  /// - Parameters:
-  ///   - tokens: The list of tokens to add to the Token Management System.
-  public mutating func set(tokens: [String]) {
-    tokenStore.set(tokens: tokens)
+    configuration.authToken = token
```

Comment:
What about thread safety? Excuse my lack of any knowledge if the answer is somehow  obvious :)

Reply:
This is property, which user won’t change often.
Also this configuration is value type - means that if it is passed somewhere, it will be copy of that structure.